### PR TITLE
Add SXC 9 security post

### PR DESCRIPTION
**Pull request by Jonne**

### CODE REVIEW

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+          "sitecore\\Commerce Business User"
+        ]
+    }
+```
+
+![Certificate authorization](/assets/images/sxc9-security/certificateauthorization.jpg))
+
+
+## Bearer token authentication
+When the client is authentication with bearer authentication the token will contains the roles of the user in its claims. When the user logs in using Identity Server, Identity Server will retrieve the users roles from the core database and return them in the token.
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: contain

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title:  "Sitecore eXperience Commerce 9 authentication and authorization"
+date: 2019-07-19 10:46
+author: jonnekats
+comments: true
+keywords: Sitecore 9, SXC9, commerce, security, identity server, authentication, authorization, explained
+categories: [Sitecore]
+---
+Authentication with SXC9 is available in het online documentation, but how does authorization work? And how can you add authorization to your own controller methods?
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: the

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+keywords: Sitecore 9, SXC9, commerce, security, identity server, authentication, authorization, explained
+categories: [Sitecore]
+---
+Authentication with SXC9 is available in het online documentation, but how does authorization work? And how can you add authorization to your own controller methods? 
+
+# Authentication
+As stated by the [SXC9 documentation](https://doc.sitecore.com/developers/91/sitecore-experience-commerce/en/authentication.html), there are two ways to authenticate with the commerce engine.
+
+## Certificate authentication
+In the case of certificate authentication a certificate thumbprint is passed to the engine in the HTTP headers. This thumprint is configured in the `config.json` file of the engine and is validated for every request.
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: thumbprint

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+## Bearer token authentication
+With bearer token authentication a token is needed to authenticate with the commerce engine. This token can be obtained from Identity Server with sitecore credentials. There are two ways to do this:
+
+### GetToken
+Identity Server provides a GetToken HTTP endpoint that accepts the sitecore credentials and returns a bearer token. This token can be used to communicate with the engine. The GetToken endpoint is mostly used by automated scripts or by the postman scripts that are part of the commerce SDK.
+
+### Authorize 
+This endpoint is used by the business tools. When the user opens the business tools, the user is redirected to the authorize endpoint of Identity Server, which shows a login page. After the user has successfuly entered credentials, the user is redirected to the business tools with the access token as a query parameters. That token can then be used by the business tools to access the commerce engine.
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: successfully

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+![Authentication overview](/assets/images/sxc9-security/overview.jpg))
+
+# Authorization
+The commerce engine determines what client can access what functionality is based on a roles and controller actions. This is configured in the `Core.ControllerMethodRoles.PolicySet-1.0.0.json` policy file, which content contains elements that look like the following:
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: "a roles and controller actions" > "roles and controller actions"?

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+      {
+        "$type": "Sitecore.Commerce.Core.ControllerMethodRolesPolicy, Sitecore.Commerce.Core",
+        "ControllerMethodRegularExpression": "/api/",
+        "AuthorizedRoles": [
+          "sitecore\\Commerce Business User"
+        ]
+      }
+```
+
+The `ControllerMethodRegularExpression` contains a regular expressions that indicates to what api routes the policy applies. The `AuthorizedRoles` states what roles are authorize to access that route. So if you need to add authorization to you own controller methods, you need to add it here.
```
Comment (robvanuden) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md]: expression

**Review by robvanuden — COMMENTED** [review]
typos

File: _posts/2019-07-19-sitecore-commerce-security-explained.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+          "sitecore\\Commerce Business User"
+        ]
+    }
+```
+
+![Certificate authorization](/assets/images/sxc9-security/certificateauthorization.jpg))
+
+
+## Bearer token authentication
+When the client is authentication with bearer authentication the token will contain the roles of the user in its claims. When the user logs in using Identity Server, Identity Server will retrieve the users roles from the core database and return them in the token.
```
Comment (MaartenKok8) [inline _posts/2019-07-19-sitecore-commerce-security-explained.md:63]: > When the client is authentication

-> authenticated

**Pull request merged**